crates/vm/levm/src/vm.rs

```diff
@@ -200,11 +200,202 @@ pub fn word_to_address(word: U256) -> Address {
+    salt: Option<U256>,
+) -> Result<VM, VMError> {
+    /*
+    Functionality should be:
+    (1) Check whether caller has enough balance to make a transfer
+    (2) Derive the new contract’s address from the caller’s address (passing in the creator account’s nonce)
+    (3) Create the new contract account using the derived contract address (changing the “world state” StateDB)
+    (4) Transfer the initial Ether endowment from caller to the new contract
+    (5) Set input data as contract’s deploy code, then execute it with EVM. The ret variable is the returned contract code
+    (6) Check for error. Or if the contract code is too big, fail. Charge the user gas then set the contract code
+    Source: https://medium.com/@hayeah/diving-into-the-ethereum-vm-part-5-the-smart-contract-creation-process-cb7b6133b855
+     */
+    let mut db_copy = db.clone();
+    let mut sender_account = match db_copy.accounts.get(&sender) {
+        Some(acc) => acc,
+        None => {
+            return Err(VMError::SenderAccountDoesNotExist);
+        }
+    }
+    .clone();
+
+    // (1)
+    if sender_account.balance < value {
+        return Err(VMError::OutOfGas); // Maybe a more personalized error
+    }
+
+    db_copy.increment_account_nonce(&sender);
+    // Check for nonce errors?
+
+    sender_account.balance -= value;
+
+    // (2)
+    let new_contract_address = match salt {
+        Some(salt) => VM::calculate_create2_address(sender, &calldata, salt),
+        None => VM::calculate_create_address(sender, sender_account.nonce),
+    };
+
+    // If address is already in db, there's an error
+    if db_copy.accounts.contains_key(&new_contract_address) {
+        return Err(VMError::AddressAlreadyOccuped);
+    }
+
+    // (3)
+    let mut created_contract = Account::new(
+        new_contract_address,
+        value,
+        calldata.clone(),
+        1,
+        Default::default(),
+    );
+    db_copy.add_account(new_contract_address, created_contract.clone());
+
+    // (4)
+    sender_account.balance -= value;
+    created_contract.balance += value;
+
+    // (5)
+    let code: Bytes = calldata.clone();
+
+    // Call the contract
+    let mut vm = VM::new(
+        Some(created_contract.address),
+        sender,
+        value,
+        code,
+        sender_account.balance,
+        block_number,
+        coinbase,
+        timestamp,
+        prev_randao,
+        chain_id,
+        base_fee_per_gas,
+        gas_price,
+        &mut db_copy,
+        block_blob_gas_used,
+        block_excess_blob_gas,
+        tx_blob_hashes,
+        secret_key,
+        None,
+    )?;
+
+    let res = vm.transact()?;
+    // Don't use a revert bc work with clones, so don't have to save previous state
+
+    let contract_code = res.output;
+
+    // (6)
+    if contract_code.len() > 24576 {
+        return Err(VMError::ContractOutputTooBig);
+    }
+    // Supposing contract code has contents
+    if contract_code[0] == 0xef {
```

Comment:
Use a constant for `0xef`.

Reply:
I named it INVALID_CONTRACT_PREFIX, it's ok?

Reply:
Yes!